apps/platform-app/src/app/[locale]/(profile)/(components)/TogglePublishButton/TogglePublishButton.tsx

```diff
@@ -13,16 +13,25 @@ import { useTranslations } from 'next-intl'
 import { publishProfile } from '../../_actions/mutations/publishProfile'
 import { unpublishProfile } from '../../_actions/mutations/unpublishProfile'
 import { DevTypeButton } from '../Filters/Buttons/DevTypeButton/DevTypeButton'
+import VerificationModal from '../VerificationModal/VerificationModal'
 
 interface TogglePublishButtonProps {
   profileId: string
   state: PublishingState
+  lastRejectionReason: string
 }
 
-export const TogglePublishButton = (props: TogglePublishButtonProps) => {
+export const TogglePublishButton = async (props: TogglePublishButtonProps) => {
```

Comment:
**suggestion**: Add `state` as a dependency in the `useEffect` dependency array to ensure it runs when `state` changes.